schemas/extractor.yml

```diff
@@ -0,0 +1,74 @@
+---
+name: extractor
+id: https://www.marda-alliance.org/extractors/extractor
+prefixes:
+    linkml: https://w3id.org/linkml/
+    schema_org: http://schema.org/
+    dublin_core: http://purl.org/dc/elements/1.1/
+imports:
+    - linkml:types
+default_range: string
+description: >-
+    This file describes the `Extractor` model created by the MaRDA extractors WG.
+
+
+classes:
+    Extractor:
+        description: >-
+            A code object or web service that when executed can read a file with specific
+            file type, or a
+            set of files with a given structure, and extract or transform the scientific
+            data
+            and/or metadata contained within the file.
+        close_mappings:
+            - schema_org:SoftwareApplication
+            - schema_org:ServiceChannel
+            - dublin_core:Software
+            - dublin_core:Service
+        attributes:
+            id:
+                identifier: true
+                slot_uri: schema_org:identifier
+                required: true
+                description: >-
+                    A unique identifier for the format within the MaRDA
+                    registry namespace, this should be a shorthand label rather than
+                    a UUID. Only lower-case alphanumeric and dash ("-") characters
+                    are permitted.
+                pattern: ^[a-z]+[a-z,0-9,-]*[a-z,0-9]+$
+            name:
+                slot_uri: schema_org:name
+                required: true
+                description: >-
+                    A recognisable name for the extractor.
```

Comment:
I'm not sure why we need both `id` and `name` here. In theory, I think `id` could just be `name-version`, generated automatically. The description of `id` is also still talking about filetypes...

Reply:
We should also clarify whether we want such granularity at this level, or whether the actual extractor should just be `yadg` with a much longer list of supported file types (though this then forces you to implement file type detection [via user-provided parameter or file contents] so that the correct parser is called when someone simply invokes `yadg` on a file).

Reply:
Well, that depends on the entrypoint, no? Assuming something like this works:
```
yadg parse biologic.mpr my_mpr_file.mpr
```

We have two options. Either we do as you want, and have `id: yadg-biologic-mpr` which has the following entrypoint:
```
entrypoint: yadg parse biologic.mpr %f
```
Note the period in `biologic.mpr`.

Or we do `id: yadg` with an additional `supported-filetypes: biologic-mpr` with an entrypoint:
```
entrypoint: yadg parse %ftype %fname
```

I think the first option might be a little bit easier to write, as it likely doesn't require *any* changes in the extractor code. The second one would necessarily involve the extractor supporting `biologic-mpr` (note the dash) as a filetype.